From 5a36afdb2f342f06001a288f744b8d4e81f52738 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:04 -0400
Subject: [virt] kvm/vfio: silence GCC warning

Message-id: <1406040016-3289-30-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86108
O-Subject: [RHEL7 PATCH v2 029/221] kvm: vfio: silence GCC warning
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Paul Bolle <pebolle@tiscali.nl>

Building vfio.o triggers a GCC warning (when building for 32 bits x86):
    arch/x86/kvm/../../../virt/kvm/vfio.c: In function 'kvm_vfio_set_group':
    arch/x86/kvm/../../../virt/kvm/vfio.c:104:22: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
      void __user *argp = (void __user *)arg;
                          ^

Silence this warning by casting arg to unsigned long.

argp's current type, "void __user *", is always casted to "int32_t
__user *". So its type might as well be changed to "int32_t __user *".

Signed-off-by: Paul Bolle <pebolle@tiscali.nl>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit e81d1ad32753cdeaef56b9bffe3b8ab7b5c776e5)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/virt/kvm/vfio.c b/virt/kvm/vfio.c
index ca4260e..b4f9507 100644
--- a/virt/kvm/vfio.c
+++ b/virt/kvm/vfio.c
@@ -101,14 +101,14 @@ static int kvm_vfio_set_group(struct kvm_device *dev, long attr, u64 arg)
  struct kvm_vfio *kv = dev->private;
  struct vfio_group *vfio_group;
  struct kvm_vfio_group *kvg;
- void __user *argp = (void __user *)arg;
+ int32_t __user *argp = (int32_t __user *)(unsigned long)arg;
  struct fd f;
  int32_t fd;
  int ret;
 
  switch (attr) {
  case KVM_DEV_VFIO_GROUP_ADD:
-  if (get_user(fd, (int32_t __user *)argp))
+  if (get_user(fd, argp))
    return -EFAULT;
 
   f = fdget(fd);
@@ -148,7 +148,7 @@ static int kvm_vfio_set_group(struct kvm_device *dev, long attr, u64 arg)
   return 0;
 
  case KVM_DEV_VFIO_GROUP_DEL:
-  if (get_user(fd, (int32_t __user *)argp))
+  if (get_user(fd, argp))
    return -EFAULT;
 
   f = fdget(fd);
-- 
1.7.1